rust/gsc/src/read_job.rs

```diff
@@ -0,0 +1,92 @@
+use axum::Json;
+use entity::{job, output_dir, output_file, output_symlink};
+use sea_orm::{
+    ColumnTrait, DatabaseConnection, DbErr, EntityTrait, ModelTrait, QueryFilter, TransactionTrait,
+};
+use std::sync::Arc;
+use tracing;
+
+use crate::types::{Dir, File, ReadJobPayload, ReadJobResponse, Symlink};
+
+#[tracing::instrument]
+pub async fn read_job(
+    Json(payload): Json<ReadJobPayload>,
+    conn: Arc<DatabaseConnection>,
+) -> Json<ReadJobResponse> {
+    // First find the hash so we can look up the exact job
+    let hash: Vec<u8> = payload.hash().into();
+
+    let result = conn
+        .as_ref()
+        .transaction::<_, ReadJobResponse, DbErr>(|txn| {
+            Box::pin(async move {
+                let Some(matching_job) = job::Entity::find()
+                    .filter(job::Column::Hash.eq(hash))
+                    .one(txn)
+                    .await?
+                else {
+                  return Ok(ReadJobResponse::NoMatch);
+                };
+
+                let output_files = matching_job
```

Comment:
can this be

```Rust
let stuff = matching_job
  .find_related(output_files::Entity)
  .find_related(output_symlink::Entity)
  .find_related(output_dir::Entity)
  .all(txn)
  .await?
```
?

I'm guessing the `.map` is the headache there?

Reply:
I wasn't able to get that to work but maybe you can give it a try later.